.jenkins/profiling.groovy

```diff
@@ -0,0 +1,123 @@
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+// This shared library is available at https://github.com/ROCmSoftwarePlatform/rocJENKINS/
+@Library('rocJenkins@pong') _
+
+// This is file for internal AMD use.
+// If you are interested in running your own Jenkins, please raise a github issue for assistance.
+
+import com.amd.project.*
+import com.amd.docker.*
+import java.nio.file.Path
+
+def runCompileCommand(platform, project, jobName, boolean debug=false)
+{
+    project.paths.construct_build_prefix()
+
+    def command = """#!/usr/bin/env bash
+            set -ex
+            hostname
+            cd ${project.paths.project_build_prefix}
+
+            # Temporary fix to remedy incorrect home directory
+            export HOME=/home/jenkins
+
+            gpuArch=`/opt/rocm/bin/rocm_agent_enumerator  | tail -n 1`
+            gfx_name="gfx90a"
+            arch_name="aldebaran"
+            logic_path="library/src/blas3/Tensile/Logic/asm_full"
+
+            mkdir tmp && cd tmp
+            git clone --no-checkout --filter=tree:0 https://github.com/ROCm/rocBLAS.git .
+            git sparse-checkout set $logic_path/$arch_name
+            git checkout
+            cd ..
+
+            python3 -m cProfile -o cprofile.prof Tensile/bin/TensileCreateLibrary \
+              $PWD/tmp/$logic_path/$arch_name \
+              _build \
+              HIP \
+              --merge-files \
+              --separate-architecture \
+              --lazy-library-loading \
+              --no-short-file-names \
+              --no-library-print-debug \
+              --code-object-version=default \
+              --cxx-compiler=hipcc \
+              --jobs=32 \
+              --library-format=msgpack \
+              --architecture=$gfx_name
+            """
+
+    try
+    {
+        platform.runCommand(this, command)
+    }
+    catch(e)
+    {
+        throw e
+    }
+
+    archiveArtifacts artifacts: 'cprofile.prof'
+}
+
+
+def runCI =
+{
+    nodeDetails, jobName ->
+
+    def prj = new rocProject('Tensile', 'Profiling')
+
+    // Define test architectures; an optional ROCm version argument is available
+    def nodes = new dockerNodes(nodeDetails, jobName, prj)
+
+    boolean formatCheck = true
```

Comment:
Remove as this is causing problems with the pipeline and isn't required for profiling.